app/Models/BeatmapSet.php

```diff
@@ -242,8 +250,18 @@ public static function search(array $params = [])
             }
         }
 
-        if (presence($query)) {
-            $matchParams[] = ['query_string' => ['query' => implode(' AND ', explode(' ', $query))]];
+        if (presence($query) != null) {
+            $matchParams[] = ['query_string' => ['query' => $query]];
+        }
+
+        if (!empty($rank)) {
```

Comment:
A reminder that `empty('0')` is true :D (though tracing back to its caller does guarantee that it's array).